src/averageAges.js

```diff
@@ -55,7 +91,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = onlyWithSon
+    ? people.filter(person => person.sex === 'f' && hasSon(person, people))
+    : people.filter(person => {
+      return person.sex === 'f' && hasChildren(person, people);
+    });
```

Comment:
використай скорочений запис стрілочний функцій
```suggestion
    : people.filter(person => (
        person.sex === 'f' && hasChildren(person, people);
    ));
```